### DISCUSSION TITLE
Commits remove CNAME file of custom domain

### DISCUSSION THREAD

**Discussion by msh855** [quarto-dev/quarto-cli#5341] [Q&A] [answered]
I have recently updated to Quarto 1.3 but it seems is creating problems when I am trying to deploy my blog. 

I am using Github pages and for deployment, I am just using the publish command in the terminal. 

I have found out that after I start using Quarto 1.3 the process for deployment [seems to be erasing the CNAME](https://github.com/msh855/TheQuantEconomist/commit/146e92b9f99ec3d80b45be67884e4d4511586830#diff-76b26edb75578663ca0fd6515ec7880ca04046bdbab298938981c9f0bd3907dd) file needed for the custom domain: 

This is causing Github to also revert back to the default url, and not my custom domain.

**Comment by dragonstyle** [comment, marked as answer]
I think this is likely what you're looking for:

https://github.com/quarto-dev/quarto-cli/discussions/3249

If you move your CNAME file next to your `_quarto.yml` file and add it as a resource:

```yaml
resources:
  - CNAME
```

It will be copied to the output directory when you build your website.

**Reply by nateraw** [reply]
Thanks - this fix worked fine for me, just was a little frustrating to track down. It would be really nice if this was included in the documentation. ❤️

As an aside, perhaps it would be nice if CNAME didn't get overridden at all. If you follow the directions from the docs right now, you end up creating a CNAME when you run `quarto publish gh-pages` (if you set up the custom domain correctly already), and then it gets deleted.

**Reply by mcanouil** [reply]
> As an aside, perhaps it would be nice if CNAME didn't get overridden at all. If you follow the directions from the docs right now, you end up creating a CNAME when you run quarto publish gh-pages (if you set up the custom domain correctly already), and then it gets deleted.

So you wouldn't be able to update the CNAME file from the source code. I don't see how this would be a good idea.

This is not limited to CNAME but to any files you want to have alongside your rendered website. This is documented at <https://quarto.org/docs/websites/website-tools.html#site-resources>.

**Reply by nateraw** [reply]
Makes sense! Thanks for the context. If I were to put a PR in to add this to the docs, what page would it be best to add it to?

**Reply by mcanouil** [reply]
To add what?

**Reply by nateraw** [reply]
a note about how when you're using the quarto publish github action with a custom domain, you need to make sure to add CNAME to the root of the repo as well as CNAME as a resource.

**Reply by mcanouil** [reply]
I suggest you open a "[documentation issue](https://github.com/quarto-dev/quarto-cli/issues/new/choose)" to express what you feel is missing.

**Reply by hamelsmu** [reply]
@mcanouil I am working with nate on some quarto projects.  Do you think a PR to the docs would be better or opening an issue?

**Reply by mcanouil** [reply]
Well, both works.
A "documentation issue" is the perfect place to discuss what could be added, how, and where.
While a PR can also do that but after the fact, meaning you might have to refactor your PR.

Note that the CNAME thing is not only for GitHub Pages, but it is true for absolutely all hosts.
In my opinion, adding a note every where or only for publishing to GitHub is wrong.

What I think is best (tell me if this would work): I would tweak or add a small paragraph and example in <https://quarto.org/docs/websites/website-tools.html#site-resources>.

For example:

> for example, when publishing to a web host, custom domain can be defined via `CNAME` and added to your site with:
>
> ```yaml
> project:
>   type: website
>   resources:
>     - CNAME
> ```
> 
> On Netlify, you can also provide `_headers` (<https://docs.netlify.com/routing/headers/>):
> 
> ```yaml
> project:
>   type: website
>   resources:
>     - _headers
> ```

**Reply by mcanouil** [reply]
I went ahead with this:

- https://github.com/quarto-dev/quarto-web/pull/1091

**Comment by msh855** [comment]
I did and the problem persists. Here is the [link](https://github.com/msh855/TheQuantEconomist) to my repo in case it helps. 

It clearly there is something that happens in the background and changes the url.

**Comment by dragonstyle** [comment]
I don't see a CNAME file checked into the root of your repository. If this isn't working / you want some advice from others that are a bit more familiar with GitHub Pages than I am, I think the discussion I linked to would be a good place to start / post additional follow up questions.
